java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/JdbcToArrowTestHelper.java

```diff
@@ -0,0 +1,250 @@
+            if(!float8Vector.isNull(j)) {
+                assertEquals(values[j], float8Vector.get(j), 0.01);
+            }
+        }
+
+        return true;
+    }
+
+    public static boolean assertFloat4VectorValues(FieldVector fx, int rowCount, float[] values){
+        Float4Vector float4Vector = ((Float4Vector)fx);
+
+        assertEquals(rowCount, float4Vector.getValueCount());
+
+        for(int j = 0; j < float4Vector.getValueCount(); j++){
+            if(!float4Vector.isNull(j)){
+                assertEquals(values[j], float4Vector.get(j), 0.01);
+            }
+        }
+
+        return true;
+    }
+
+    public static boolean assertTimeVectorValues(FieldVector fx, int rowCount, long[] values){
+        TimeMilliVector timeMilliVector = ((TimeMilliVector)fx);
+
+        assertEquals(rowCount, timeMilliVector.getValueCount());
+
+        for(int j = 0; j < timeMilliVector.getValueCount(); j++){
+            if(!timeMilliVector.isNull(j)){
+                assertEquals(values[j], timeMilliVector.get(j));
+            }
+        }
+
+        return true;
+    }
+
+    public static boolean assertDateVectorValues(FieldVector fx, int rowCount, long[] values){
+        DateMilliVector dateMilliVector = ((DateMilliVector)fx);
+
+        assertEquals(rowCount, dateMilliVector.getValueCount());
+
+        for(int j = 0; j < dateMilliVector.getValueCount(); j++){
+            if(!dateMilliVector.isNull(j)){
+                assertEquals(values[j], dateMilliVector.get(j));
+            }
+        }
+
+        return true;
+    }
+
+    public static boolean assertTimeStampVectorValues(FieldVector fx, int rowCount, long[] values){
+        TimeStampVector timeStampVector = ((TimeStampVector)fx);
+
+        assertEquals(rowCount, timeStampVector.getValueCount());
+
+        for(int j = 0; j < timeStampVector.getValueCount(); j++){
+            if(!timeStampVector.isNull(j)){
+                assertEquals(values[j], timeStampVector.get(j));
+            }
+        }
+
+        return true;
+    }
+
+    public static boolean assertVarBinaryVectorValues(FieldVector fx, int rowCount, byte[][] values){
+        VarBinaryVector varBinaryVector =((VarBinaryVector) fx);
+
+        assertEquals(rowCount, varBinaryVector.getValueCount());
+
+        for(int j = 0; j < varBinaryVector.getValueCount(); j++){
+            if(!varBinaryVector.isNull(j)){
+                assertEquals(hashArray(values[j]), hashArray(varBinaryVector.get(j)));
+            }
+        }
+
+        return true;
+    }
+
+    public static boolean assertVarcharVectorValues(FieldVector fx, int rowCount, byte[][] values) {
+        VarCharVector varCharVector = ((VarCharVector)fx);
+
+        assertEquals(rowCount, varCharVector.getValueCount());
+
+        for(int j = 0; j < varCharVector.getValueCount(); j++){
+            if(!varCharVector.isNull(j)){
+                assertEquals(hashArray(values[j]), hashArray(varCharVector.get(j)));
```

Comment:
why comparing the hash instead of the array content? (since the hash is based on the array content anyway...)